dds/DCPS/RTPS/TypeLookup.idl

```diff
@@ -0,0 +1,87 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include <dds/DCPS/XTypes/TypeObject.idl>
+#include <dds/DCPS/RTPS/RtpsRpc.idl>
+
+module OpenDDS {
+  module XTypes {
+
+    const long DDS_RETCODE_OK = 0;
+
+    // As per chapter 7.6.3.3.3 of XTypes spec
+    // Issue DDSXTY14-30:
+    // - These use XCDR2 and default to appendable.
+    // - The @RPCRequestType/@RPCReplyType structs are final.
+
+    // Used in TypeLookup_Call and TypeLookup_Return
+    const long TypeLookup_getTypes_HashId = 0x018252d3;
+    const long TypeLookup_getDependencies_HashId = 0x05aafb31;
+
+    @extensibility(MUTABLE)
+    struct TypeLookup_getTypes_In {
+      @hashid TypeIdentifierSeq type_ids;
+    };
+
+    @extensibility(MUTABLE)
+    struct TypeLookup_getTypes_Out {
+      @hashid TypeIdentifierTypeObjectPairSeq types;
+      @hashid TypeIdentifierPairSeq complete_to_minimal;
+    };
+
+    union TypeLookup_getTypes_Result switch(long) {
+      case DDS_RETCODE_OK:
+        TypeLookup_getTypes_Out result;
+    };
+
+    typedef sequence<octet, 32> OctetSeq32;
+
+    @extensibility(MUTABLE)
+    struct TypeLookup_getTypeDependencies_In {
+      @hashid TypeIdentifierSeq type_ids;
+      @hashid OctetSeq32 continuation_point;
+    };
+
+    @extensibility(MUTABLE)
+    struct TypeLookup_getTypeDependencies_Out {
+      @hashid TypeIdentifierWithSizeSeq dependent_typeids;
+      @hashid OctetSeq32 continuation_point;
+    };
+
+    union TypeLookup_getTypeDependencies_Result switch(long) {
+      case DDS_RETCODE_OK:
+        TypeLookup_getTypeDependencies_Out result;
+    };
+
+    union TypeLookup_Call switch(long) {
+      case TypeLookup_getTypes_HashId:
+        TypeLookup_getTypes_In getTypes;
+
+      case TypeLookup_getDependencies_HashId:
+        TypeLookup_getTypeDependencies_In getTypeDependencies;
+    };
+
+    @RPCRequestType
```

Comment:
This annotation isn't defined by either IDL compiler, so`opendds_idl` will warn about this (tao_idl is set to ignore unknown annotations). We should comment them out or define the annotation, which can be done by adding `@annotation RPCRequestType {};` somewhere above here.

Reply:
Done.